crates/storage/src/prefix.rs

```diff
@@ -7,17 +7,25 @@ use {
     std::{borrow::Cow, marker::PhantomData},
 };
 
-pub struct Prefix<K, T, C>
+pub trait AsKeysIterator {}
+pub trait AsRangeIterator {}
+
+pub struct KeysIterator;
+impl AsKeysIterator for KeysIterator {}
+
+pub struct RangeIterator;
+impl AsRangeIterator for RangeIterator {}
+impl AsKeysIterator for RangeIterator {}
```

Comment:
Similar to the type state builder, these two structures with attached traits allow to change the functions that can be used in the PrefixPath. This is because in the case of the Set is possible to use both the keys and range functions, where range returned a record with the value = Empty (that is not needed at all). When the Set returns the PrefixPath, it creates it with the KeysIterator, while a map with the RangeIterator. The methods for iterating the keys are implemented for both the RangeIterator and the KeysIterator, while those for iterating Range and Values ​​are implemented only for the RangeIterator